# THYROID DISEASE DATASET PREPARATION & CLEANING

In [1]:
# Import libraries that will assist in cleaning

# Pandas for data manipulation and numpy for numeric operations

import pandas as pd
import numpy as np

In [2]:
# Load the thyroid dataset

data1 = pd.read_csv('/Users/RyanMburu/Desktop/DS Projects/Thyroid Disease Detector/hypothyroid.csv')

# Print out first 5 rows
data1.head()

age sex on thyroxine query on thyroxine on antithyroid medication sick  \
0  41   F            f                  f                         f    f   
1  23   F            f                  f                         f    f   
2  46   M            f                  f                         f    f   
3  70   F            t                  f                         f    f   
4  70   F            f                  f                         f    f   

  pregnant thyroid surgery I131 treatment query hypothyroid  ... TT4 measured  \
0        f               f              f                 f  ...            t   
1        f               f              f                 f  ...            t   
2        f               f              f                 f  ...            t   
3        f               f              f                 f  ...            t   
4        f               f              f                 f  ...            t   

   TT4 T4U measured   T4U FTI measured  FTI TBG measured TBG referral source  \
0  125            t  1.14            t  109            f   ?            SVHC   
1  102            f     ?            f    ?            f   ?           other   
2  109            t  0.91            t  120            f   ?           other   
3  175            f     ?            f    ?            f   ?           other   
4   61            t  0.87            t   70            f   ?             SVI   

  binaryClass  
0           P  
1           P  
2           P  
3           P  
4           P  

[5 rows x 30 columns]

In [4]:
# Print out last 5 rows
data1.tail()

age sex on thyroxine query on thyroxine on antithyroid medication sick  \
3767  30   F            f                  f                         f    f   
3768  68   F            f                  f                         f    f   
3769  74   F            f                  f                         f    f   
3770  72   M            f                  f                         f    f   
3771  64   F            f                  f                         f    f   

     pregnant thyroid surgery I131 treatment query hypothyroid  ...  \
3767        f               f              f                 f  ...   
3768        f               f              f                 f  ...   
3769        f               f              f                 f  ...   
3770        f               f              f                 f  ...   
3771        f               f              f                 f  ...   

     TT4 measured  TT4 T4U measured   T4U FTI measured  FTI TBG measured TBG  \
3767            f    ?            f     ?            f    ?            f   ?   
3768            t  124            t  1.08            t  114            f   ?   
3769            t  112            t  1.07            t  105            f   ?   
3770            t   82            t  0.94            t   87            f   ?   
3771            t   99            t  1.07            t   92            f   ?   

     referral source binaryClass  
3767           other           P  
3768             SVI           P  
3769           other           P  
3770             SVI           P  
3771           other           P  

[5 rows x 30 columns]

In [3]:
# Gather number of rows and columns present

data1.shape

(3772, 30)

The Hypothyroid dataset has 3,770 rows of data with 30 columns